src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -111,6 +111,30 @@ internal static void GetUnderlyingTypeChain(NamedTypeSymbol underlyingTupleType,
             }
         }
 
+        /// <summary>
+        /// Gets flattened type arguments of the underlying type
+        /// which correspond to the types of the tuple elements left-to-right
+        /// </summary>
+        internal static void GetElementTypes(NamedTypeSymbol underlyingTupleType, ArrayBuilder<TypeSymbol> tupleElementTypes)
```

Comment:
We also need to handle cases like the following (for compat reasons) - with underlying tuple type used directly. 
```
voif Foo(ValueType<int, int> x) {};

foo((1,1));
```